### PR TITLE
Add `ord decode` command

### PR DESCRIPTION
**Pull request by casey**
Decodes a transaction from a file, useful for debugging inscription parsing.

### CODE REVIEW

**Comment by rot13maxi** [comment]
yes please. from a file or stdin would be great. I'd love to be able to chuck transactions from the mempool through this

File: src/subcommand/decode.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+use super::*;
+
+#[derive(Serialize, Eq, PartialEq, Deserialize, Debug)]
+pub struct Output {
+  pub inscriptions: Vec<Inscription>,
+}
+
+#[derive(Debug, Parser)]
+pub(crate) struct Decode {
+  transaction: PathBuf,
```
Comment (rot13maxi) [inline src/subcommand/decode.rs:10]: what would you think about making a flag to read from stdin? would be useful for doing things like running mempool transactions through it

Comment (casey) [reply src/subcommand/decode.rs]: Done!

File: tests/decode.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  assert_eq!(
+    CommandBuilder::new("decode transaction.bin")
+      .write("transaction.bin", buffer)
+      .run_and_deserialize_output::<Output>(),
+    Output {
+      inscriptions: vec![Inscription {
+        body: Some(vec![0, 1, 2, 3]),
+        content_type: Some(vec![
+          116, 101, 120, 116, 47, 112, 108, 97, 105, 110, 59, 99, 104, 97, 114, 115, 101, 116, 61,
```
Comment (raphjaph) [inline tests/decode.rs:52]: please make more readable

File: tests/decode.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  assert_eq!(
+    CommandBuilder::new("decode")
+      .stdin(buffer)
+      .run_and_deserialize_output::<Output>(),
+    Output {
+      inscriptions: vec![Inscription {
+        body: Some(vec![0, 1, 2, 3]),
+        content_type: Some(vec![
+          116, 101, 120, 116, 47, 112, 108, 97, 105, 110, 59, 99, 104, 97, 114, 115, 101, 116, 61,
+          117, 116, 102, 45, 56
```
Comment (raphjaph) [inline tests/decode.rs:104]: same as above

**Review by raphjaph — CHANGES_REQUESTED** [review]
small nits

**Review by raphjaph — APPROVED** [review]
lgtm!

**Pull request merged**